sdk/search/search-documents/src/serviceUtils.ts

```diff
@@ -519,8 +531,75 @@ export function generatedVectorSearchVectorizerToPublicVectorizer(
       };
       return vectorizer;
     }
+    case "aiServicesVision": {
+      const generatedVisionVectorizer = generatedVectorizer as GeneratedAIServicesVisionVectorizer;
+      const { aIServicesVisionParameters: generatedParameters } = generatedVisionVectorizer;
+      const aIServicesVisionParameters = generatedParameters
+        ? {
+            ...generatedParameters,
+            modelVersion: generatedParameters.modelVersion ?? undefined,
+            resourceUri: generatedParameters.resourceUri,
+            authIdentity: convertSearchIndexerDataIdentityToPublic(
+              generatedParameters.authIdentity,
+            ),
+          }
+        : undefined;
+      const vectorizer: AIServicesVisionVectorizer = {
+        ...generatedVisionVectorizer,
+        aIServicesVisionParameters,
+      };
+      return vectorizer;
+    }
+    case "aml": {
+      const generatedAMLVectorizer = generatedVectorizer as GeneratedAMLVectorizer;
+
+      const vectorizer: AzureMachineLearningVectorizer = {
+        ...generatedAMLVectorizer,
+        amlParameters:
+          generatedAzureMachineLearningVectorizerParametersToPublicAzureMachineLearningVectorizerParameters(
+            generatedAMLVectorizer.aMLParameters,
+          ),
+      };
+
+      return vectorizer;
+    }
     default:
-      throw Error("Unsupported vectorizer");
+      logger.warning(`Unsupported vectorizer kind: ${(generatedVectorizer as any).kind}`);
+      return generatedVectorizer as any;
```

Comment:
If it's possible, could we remove the default clause entirely? Then, when we add a new variant to the union TypeScript will let us know that not all code paths return a value and we can let the compiler help out pointing out areas where we need to handle the new variant. Let me know if that doesn't work out!

Reply:
Done!